src/types.jl

```diff
@@ -97,13 +52,416 @@ for Model in [:NeuralNetworkClassifier, :ImageClassifier]
+using MLJ
+using Flux
+import RDatasets
+
+using Random
+Random.seed!(123)
+
+MLJ.color_off()
+
+using Plots
+pyplot(size=(600, 300*(sqrt(5)-1)));
+```
+This is a very basic example, using a default builder and no standardization.
+For a more advance illustration, see [`NeuralNetworkRegressor`](@ref) or [`ImageClassifier`](@ref). First, we can load the data:
+```julia
+iris = RDatasets.dataset("datasets", "iris");
+y, X = unpack(iris, ==(:Species), colname -> true, rng=123);
+NeuralNetworkClassifier = @load NeuralNetworkClassifier
+clf = NeuralNetworkClassifier()
+```
+Next, we can train the model:
+```julia
+import Random.seed!; seed!(123)
+mach = machine(clf, X, y)
+fit!(mach)
+```
+We can train the model in an incremental fashion with the `optimizer_changes_trigger_retraining` flag set to false (which is by default). Here, we change the number of iterations and the learning rate of the optimiser:
+```julia
+clf.optimiser.eta = clf.optimiser.eta * 2
+clf.epochs = clf.epochs + 5
+
+# note that if the optimizer_changes_trigger_retraining flag was set to true
+# the model would be completely retrained from scratch because the optimizer was
+# updated
+fit!(mach, verbosity=2);
+```
+We can inspect the mean training loss using the `cross_entropy` function:
+```julia
+
+training_loss = cross_entropy(predict(mach, X), y) |> mean
+
+```
+And we can access the Flux chain (model) using `fitted_params`:
+```julia
+chain = fitted_params(mach).chain
+```
+Finally, we can see how the out-of-sample performance changes over time, using the `learning_curve` function
+```julia
+r = range(clf, :epochs, lower=1, upper=200, scale=:log10)
+curve = learning_curve(clf, X, y,
+                     range=r,
+                     resampling=Holdout(fraction_train=0.7),
+                     measure=cross_entropy)
+using Plots
+plot(curve.parameter_values,
+     curve.measurements,
+     xlab=curve.parameter_name,
+     xscale=curve.parameter_scale,
+     ylab = "Cross Entropy")
+
+savefig("iris_history.png")
+```
+See also
+[`ImageClassifier`](@ref)
+"""
+NeuralNetworkClassifier
+
+"""
+$(MMI.doc_header(ImageClassifier))
+
+`ImageClassifier`: A neural network model for making probabilistic
+"predictions of a `GrayImage` target, given a table of `Continuous` features.
+
+# Training data
+
+In MLJ or MLJBase, bind an instance `model` to data with
+mach = machine(model, X, y)
+Where
+- `X`: is any `AbstractVector` of input features (eg, a `DataFrame`) whose items
+  are of scitype `GrayImage`; check the scitype with `scitype(X)`
+- `y`: is the target, which can be any `AbstractVector` whose element
+  scitype is `Multiclass` or `OrderedFactor` with `n_out` classes;
+  check the scitype with `scitype(y)`
+
+
+# Hyper-parameters
+
+- `builder=MLJFlux.Short()`: An MLJFlux builder that constructs a neural network. Possible `builders` include: `Linear`, `Short`, and `MLP`. You can construct your own builder using the `@builder` macro, see examples for further information.
```

Comment:
These builders are unlikely to be useful for images. And #208 will change the default. So let's do:
```suggestion
- `builder`: An MLJFlux builder that constructs the neural network. 
   The fallback builds a depth-16 VGG architecture adapted to the image 
   size and number of target classes, with no batch normalisation; see the
   Metalhead.jl documentation for details. See the example below for a 
   user-specified builder.
```